scipy/stats/_multivariate.py

```diff
@@ -4563,12 +4564,32 @@ def _entropy(self, dim, df=1, shape=1):
             return multivariate_normal(None, cov=shape).entropy()
 
         shape_info = _PSD(shape)
-        halfsum = 0.5 * (dim + df)
-        half_df = 0.5 * df
-        return (-gammaln(halfsum) + gammaln(half_df)
+        shape_term = 0.5 * shape_info.log_pdet
+
+        def regular(dim, df):
+            halfsum = 0.5 * (dim + df)
+            half_df = 0.5 * df
+            return (
+                -gammaln(halfsum) + gammaln(half_df)
                 + 0.5 * dim * np.log(df * np.pi) + halfsum
                 * (psi(halfsum) - psi(half_df))
-                + 0.5 * shape_info.log_pdet)
+                + shape_term
+            )
+
+        def asymptotic(dim, df):
+            # Formula from Wolfram Alpha:
+            # "asymptotic expansion -gammaln((m+d)/2) + gammaln(d/2) + (m*log(d*pi))/2
+            #  + ((m+d)/2) * (digamma((m+d)/2) - digamma(d/2))"
+            return (
+                dim * norm._entropy() + dim / df
+                - dim * (dim - 2) * df**-2.0 / 4
+                + dim**2 * (dim - 2) * df**-3.0 / 6
+                + dim * (-3 * dim**3 + 8 * dim**2 - 8) * df**-4.0 / 24
+                + shape_term
+            )
+
+        threshold = dim * 100 * 4 / (np.log10(dim) + 1)
```

Comment:
It looks like you've found a good heuristic, but I'd be more ready to merge this if you were to instead add the fifth-order terms to the asymptotic expansion. That would probably create overlap between the regions where each expression is accurate, so you wouldn't need a heuristic.

![image](https://github.com/scipy/scipy/assets/6570539/e0b2c518-0126-4d50-bc61-c97465fcb732)

Reply:
Even if I add the fifth order term we would still need a threshold right?

Reply:
Yes. I suppose we could call a threshold a heuristic, too, but it's pretty standard practice. I think we have use a fixed threshold in all of the other asymptotic expansion PRs.

Reply:
Right, so that means using a fifth order term would enable us to specify a fixed threshold?

Reply:
Thanks!

Reply:
I hoped that adding the extra term would extend the accurate portion of the asymptotic approximation a lot further to the left. Oh well. 

> Should we only consider dim to be up till a specific value and ignore higher ones?

I think that is safe because currently `multivariate_t` accepts only dense arrays, so the memory required scales with `dim**2` right? Since no bug reports have even been filed about `multivariate_t` entropy precision, I am not sure whether anyone will make use of this extra precision, let alone users with a petabyte of RAM. If that's right, I think it's OK to stop looking beyond `dim > 1e7`.

Reply:
Sounds good.